js/src/immersive-reader-sdk.ts

```diff
@@ -5,7 +5,18 @@ import { renderButtons } from './renderButtons';
 import { close, launchAsync } from './launchAsync';
 
 window.addEventListener('load', () => {
+    if (!('Promise' in window)) {
+        dynamicallyLoadScript('https://contentstorage.onenote.office.net/onenoteltir/permanent-static-resources/promise-polyfill.min.js');
```

Comment:
let's get feedback from others on the team to discuss any problems that may arise from automatically adding scripts